src/Helpers/Tokens/TokenVerifier.php

```diff
@@ -0,0 +1,163 @@
+<?php
+declare(strict_types=1);
+
+namespace Auth0\SDK\Helpers\Tokens;
+
+use Auth0\SDK\Exception\InvalidTokenException;
+
+/**
+ * Class TokenVerifier, a generic JWT verifier.
+ * For verifying OIDC-compliant ID tokens, use Auth0\SDK\Helpers\Tokens\IdTokenVerifier
+ *
+ * @package Auth0\SDK\Helpers\Tokens
+ */
+class TokenVerifier
+{
+
+    /**
+     * Token issuer base URL expected.
+     *
+     * @var string
+     */
+    protected $issuer;
+
+    /**
+     * Token audience expected.
+     *
+     * @var string
+     */
+    protected $audience;
+
+    /**
+     * Token signature verifier.
+     *
+     * @var SignatureVerifier
+     */
+    private $verifier;
+
+    /**
+     * Clock tolerance for time-base token checks in seconds.
+     *
+     * @var integer
+     */
+    protected $leeway = 60;
+
+    /**
+     * TokenVerifier constructor.
+     *
+     * @param string            $issuer   Token issuer base URL expected.
+     * @param string            $audience Token audience expected.
+     * @param SignatureVerifier $verifier Token signature verifier.
+     */
+    public function __construct(string $issuer, string $audience, SignatureVerifier $verifier)
+    {
+        $this->issuer   = $issuer;
+        $this->audience = $audience;
+        $this->verifier = $verifier;
+    }
+
+    /**
+     * Set a new leeway time for all token checks.
+     *
+     * @param integer $newLeeway New leeway time for class instance.
+     *
+     * @return void
+     */
+    public function setLeeway(int $newLeeway) : void
+    {
+        $this->leeway = $newLeeway;
+    }
+
+    /**
+     * Verifies and decodes an JWT.
+     *
+     * @param string $token   Raw JWT string.
+     * @param array  $options Options to adjust the verification. Can be:
+     *      - "nonce" to check the nonce contained in the token (recommended).
+     *      - "max_age" to check the auth_time of the token.
+     *      - "time" Unix timestamp to use as the current time for exp, iat, and auth_time checks. Used for testing.
```

Comment:
Question: is this typically something we want to highlight as an option in the docs and right there in the editor? Wouldn't it make it more visible and thus more likely someone could use this to manipulate the validator?

Reply:
Either way, if the documentation is to remain, it doesn't look like `iat` and `auth_time` are being checked here. Looks like those checks have been left in the subclass.